lib/reporting/cloudwatch_client.rb

```diff
@@ -205,9 +225,13 @@ def has_more_results?(size)
     # @param [Array<Array<Aws::CloudWatchLogs::Types::ResultField>>] results
     # @return [Array<Hash>]
     def parse_results(results)
-      results.map do |row|
-        row.map { |cell| [cell.field, cell.value] }.to_h.tap do |h|
-          h.delete('@ptr') # just noise
+      if count_distinct
+        [results.size]
```

Comment:
This is going to make the math wrong.

Example: you're counting events by user_id. 


user A signs in to the app every day for a month. If the query gets sliced  by week, user A will have a "count" in each of those weeks (we'll call it a 4 week month), so they're in the final count 4x when it should be one.

I think the "right" math here would be basically to merge a giant set of the user_id keys and take its size at the end

Reply:
Ah, yeah, that's a good point, I hadn't considered that.

Was hoping to avoid working with large data, but seems it must be so!

Reply:
I revised the approach in 5fcad87 to accumulate a set of unique values and generate the size from that.

(Aside: I chose a hash over `Set` after doing some small local benchmarking showing that it performs a little better for insertion)